### ISSUE TITLE
Can't build godot on linux

### ISSUE THREAD

**Issue by toby3d** [godotengine/godot#10601]
Latest (05a6785) commit from master, Linux Mint XFCE 18.2 (64bit):
```bash
$ scons platform=x11
scons: Reading SConscript files ...
Could not detect MinGW. Ensure its binaries are in your PATH or that MINGW32_PREFIX or MINGW64_PREFIX are properly defined.
Enabling ALSA
Enabling PulseAudio
('translations: ', ['/home/toby3d/git/godot/editor/translations/ar.po', '/home/toby3d/git/godot/editor/translations/bg.po', '/home/toby3d/git/godot/editor/translations/bn.po', '/home/toby3d/git/godot/editor/translations/ca.po', '/home/toby3d/git/godot/editor/translations/cs.po', '/home/toby3d/git/godot/editor/translations/da.po', '/home/toby3d/git/godot/editor/translations/de.po', '/home/toby3d/git/godot/editor/translations/de_CH.po', '/home/toby3d/git/godot/editor/translations/el.po', '/home/toby3d/git/godot/editor/translations/es.po', '/home/toby3d/git/godot/editor/translations/es_AR.po', '/home/toby3d/git/godot/editor/translations/fa.po', '/home/toby3d/git/godot/editor/translations/fi.po', '/home/toby3d/git/godot/editor/translations/fr.po', '/home/toby3d/git/godot/editor/translations/hu.po', '/home/toby3d/git/godot/editor/translations/id.po', '/home/toby3d/git/godot/editor/translations/it.po', '/home/toby3d/git/godot/editor/translations/ja.po', '/home/toby3d/git/godot/editor/translations/ko.po', '/home/toby3d/git/godot/editor/translations/nb.po', '/home/toby3d/git/godot/editor/translations/nl.po', '/home/toby3d/git/godot/editor/translations/pl.po', '/home/toby3d/git/godot/editor/translations/pr.po', '/home/toby3d/git/godot/editor/translations/pt_BR.po', '/home/toby3d/git/godot/editor/translations/pt_PT.po', '/home/toby3d/git/godot/editor/translations/ru.po', '/home/toby3d/git/godot/editor/translations/sk.po', '/home/toby3d/git/godot/editor/translations/sl.po', '/home/toby3d/git/godot/editor/translations/th.po', '/home/toby3d/git/godot/editor/translations/tr.po', '/home/toby3d/git/godot/editor/translations/ur_PK.po', '/home/toby3d/git/godot/editor/translations/zh_CN.po', '/home/toby3d/git/godot/editor/translations/zh_HK.po', '/home/toby3d/git/godot/editor/translations/zh_TW.po'])
('fonts: ', ['/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSans.ttf', '/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSansArabic.ttf', '/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSansFallback.ttf', '/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSansHebrew.ttf', '/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSansJapanese.ttf', '/home/toby3d/git/godot/editor/../thirdparty/fonts/DroidSansThai.ttf', ['/home/toby3d/git/godot/editor/../thirdparty/fonts/source_code_pro.otf']])
scons: done reading SConscript files.
scons: Building targets ...
[ 35%] Compiling ==> editor/editor_icons.gen.cpp
editor/editor_icons.gen.cpp:1:26: fatal error: editor_icons.h: No such file or directory
compilation terminated.
scons: *** [editor/editor_icons.gen.x11.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
That's a transient issue, you can either clean up all your build artifacts or just `find -name "*.gen.*" -delete`.

**Comment by toby3d** [comment]
@akien-mga Okay, thanks. But what about android templates?
```bash
$ scons platform=android target=release
scons: Reading SConscript files ...
Could not detect MinGW. Ensure its binaries are in your PATH or that MINGW32_PREFIX or MINGW64_PREFIX are properly defined.
Building for Android (armv7) (with NEON)
scons: done reading SConscript files.
scons: Building targets ...
[ 16%] Compiling ==> modules/gdscript/gd_editor.cpp
modules/gdscript/gd_editor.cpp:2488:25: error: out-of-line definition of 'complete_code' does not match any declaration in 'GDScriptLanguage'
Error GDScriptLanguage::complete_code(const String &p_code, const String &p_base_path, Object *p_owner, List<String> *r_options, String &r_call_hint) {
                        ^~~~~~~~~~~~~
1 error generated.
scons: *** [modules/gdscript/gd_editor.android.opt.armv7.neon.o] Error 1
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
Thanks, I just fixed that one with 3ea0943f.
